plugins/defaultplugins/ifplugin/interface_config_test.go

```diff
@@ -0,0 +1,985 @@
+	"github.com/ligato/cn-infra/logging/logrus"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/af_packet"
+	dhcp_api "github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/dhcp"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/interfaces"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/ip"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/memif"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/tap"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/tapv2"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/vpe"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/common/bin_api/vxlan"
+	if_api "github.com/ligato/vpp-agent/plugins/defaultplugins/common/model/interfaces"
+	"github.com/ligato/vpp-agent/plugins/defaultplugins/ifplugin"
+	"github.com/ligato/vpp-agent/tests/vppcallmock"
+	. "github.com/onsi/gomega"
+)
+
+/* Interface configurator init and close */
+
+// Test init function
+func TestInterfaceConfiguratorInit(t *testing.T) {
+	var err error
+	// Setup
+	RegisterTestingT(t)
+	ctx := &vppcallmock.TestCtx{
+		MockVpp: &mock.VppAdapter{},
+	}
+	connection, _ := core.Connect(ctx.MockVpp)
+	defer connection.Disconnect()
+	plugin := &ifplugin.InterfaceConfigurator{}
+	ifVppNotifChan := make(chan govppapi.Message, 100)
+	// Reply set
+	ctx.MockVpp.MockReply(&memif.MemifSocketFilenameDetails{
+		SocketID:       1,
+		SocketFilename: []byte("test-socket-filename"),
+	})
+	ctx.MockVpp.MockReply(&vpe.ControlPingReply{})
+	// Test init
+	err = plugin.Init("test-plugin", logging.ForPlugin("test-log", logrus.NewLogRegistry()), connection,
+		nil, ifVppNotifChan, 0, false)
+	Expect(err).To(BeNil())
+	Expect(plugin.IsSocketFilenameCached("test-socket-filename")).To(BeTrue())
+	// Test close
+	err = plugin.Close()
+	Expect(err).To(BeNil())
+}
+
+// Test init function
+func TestInterfaceConfiguratorDHCPNotifications(t *testing.T) {
+	var err error
+	// Setup
+	_, connection, plugin := ifTestSetup(t)
+	defer ifTestTeardown(connection, plugin)
+	// Register
+	plugin.GetSwIfIndexes().RegisterName("if1", 1, nil)
+	plugin.GetSwIfIndexes().RegisterName("if2", 2, nil)
+	// Test DHCP notifications
+	dhcpIpv4 := &dhcp_api.DhcpComplEvent{
+		HostAddress:   net.ParseIP("10.0.0.1"),
+		RouterAddress: net.ParseIP("10.0.0.2"),
+		HostMac: func(mac string) []byte {
+			parsed, _ := net.ParseMAC(mac)
+			return parsed
+		}("7C:4E:E7:8A:63:68"),
+		Hostname: []byte("if1"),
+		IsIpv6:   0,
+	}
+	dhcpIpv6 := &dhcp_api.DhcpComplEvent{
+		HostAddress:   net.ParseIP("fd21:7408:186f::/48"),
+		RouterAddress: net.ParseIP("2001:db8:a0b:12f0::1/48"),
+		HostMac: func(mac string) []byte {
+			parsed, err := net.ParseMAC(mac)
+			Expect(err).To(BeNil())
+			return parsed
+		}("7C:4E:E7:8A:63:68"),
+		Hostname: []byte("if2"),
+		IsIpv6:   1,
+	}
+	plugin.DhcpChan <- dhcpIpv4
+	time.Sleep(1 * time.Second)
```

Comment:
This should not be necessary since we use Eventually, should it?

Reply:
forgot to remove, sorry